pkgs/development/compilers/dmd/sourcebuild.nix

```
@@ -0,0 +1,168 @@
+{ stdenv, lib, fetchFromGitHub
+, makeWrapper, unzip, which, writeTextFile
+, curl, tzdata, gdb, Foundation, git, callPackage
+, targetPackages, fetchpatch, bash
+, HOST_DMD
+, version
+, dmdSha256
+, druntimeSha256
+, phobosSha256
+}:
+
+let
+  dmdConfFile = writeTextFile {
+    name = "dmd.conf";
+    text = (lib.generators.toINI {} {
+      Environment = {
+        DFLAGS = ''-I@out@/include/dmd -L-L@out@/lib -fPIC ${lib.optionalString (!targetPackages.stdenv.cc.isClang) "-L--export-dynamic"}'';
+      };
+    });
+  };
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "dmd";
+  inherit version;
+
+  enableParallelBuilding = true;
+
+  srcs = [
+  (fetchFromGitHub {
+    owner = "dlang";
+    repo = "dmd";
+    rev = "v${version}";
+    sha256 = dmdSha256;
+    name = "dmd";
+  })
+  (fetchFromGitHub {
+    owner = "dlang";
+    repo = "druntime";
+    rev = "v${version}";
+    sha256 = druntimeSha256;
+    name = "druntime";
+  })
+  (fetchFromGitHub {
+    owner = "dlang";
+    repo = "phobos";
+    rev = "v${version}";
+    sha256 = phobosSha256;
+    name = "phobos";
+  })
+  ];
+
+  sourceRoot = ".";
+
+  # https://issues.dlang.org/show_bug.cgi?id=19553
+  hardeningDisable = [ "fortify" ];
+
+  postUnpack = ''
+    patchShebangs .
```

```suggestion
  postUnpack = ''
    patchShebangs .
```
move this to postPatch. Which files exactly need patching?

---

I don't know, so left it as-is (but moved to `patchPhase` nonetheless)